### PR TITLE
Fix #1033: Template detail endpoint

### LINKED ISSUE #1033: Template detail endpoint
Expose an endpoint providing template details `/api/admin/template`.
The endpoint should not be exported into the swagger and should be authenticated.

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

File: enrollment-server/src/main/java/com/wultra/app/enrollmentserver/configuration/OpenApiConfiguration.java [hunk 1]
```diff
@@ -49,16 +50,28 @@
...
                 .packagesToScan(packages)
+                .packagesToExclude(PACKAGE_ADMIN)
+                .build();
+    }
+
+    @Bean
+    @ConditionalOnProperty(value = "enrollment-server.admin.enabled", havingValue = "true")
+    public GroupedOpenApi adminApiGroup() {
+        return GroupedOpenApi.builder()
+                .group("enrollment-server-admin")
```
Comment (banterCZ) [inline enrollment-server/src/main/java/com/wultra/app/enrollmentserver/configuration/OpenApiConfiguration.java:73]: Admin controller is not shown on the default swagger page, but you may see it if you want to.

File: enrollment-server/src/main/java/com/wultra/app/enrollmentserver/controller/api/admin/AdminController.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+@RestController
+@RequestMapping(value = "api/admin")
+@AllArgsConstructor
+@Slf4j
+public class AdminController {
+
+    private final OperationTemplateService operationTemplateService;
+
+    @GetMapping("/template")
+    public ObjectResponse<List<TemplateDetailResponse>> templates() {
```
Comment (banterCZ) [inline enrollment-server/src/main/java/com/wultra/app/enrollmentserver/controller/api/admin/AdminController.java:49]: Is it needed to introduce paging? I do not expect too many templates but I may be wrong.

File: enrollment-server/src/main/java/com/wultra/app/enrollmentserver/configuration/SecurityConfig.java [hunk 3]
```diff
@@ -32,14 +46,64 @@
...
+        return new InMemoryUserDetailsManager(user);
+    }
+
+    @ConditionalOnProperty(value = "enrollment-server.auth-type", havingValue = "BASIC_HTTP" )
+    @Bean
+    public PasswordEncoder passwordEncoder() {
+        return PasswordEncoderFactories.createDelegatingPasswordEncoder();
+    }
+
+    enum AuthType {
```
Comment (romanstrobl) [inline enrollment-server/src/main/java/com/wultra/app/enrollmentserver/configuration/SecurityConfig.java:99]: There is a comment only for one authentication type.

File: enrollment-server/src/main/java/com/wultra/app/enrollmentserver/controller/api/admin/AdminController.java (hunk 2 above)
Comment (romanstrobl) [reply enrollment-server/src/main/java/com/wultra/app/enrollmentserver/controller/api/admin/AdminController.java:49]: I think it's not necessary.

File: enrollment-server/src/main/resources/application.properties [hunk 4]
```diff
@@ -68,6 +68,16 @@ powerauth.service.security.clientSecret=
 enrollment-server.mtoken.enabled=true
 enrollment-server.inbox.enabled=true
 enrollment-server.activation-spawn.enabled=false
+enrollment-server.admin.enabled=false
+enrollment-server.auth-type=NONE
+
+#Basic HTTP Settings
```
Comment (romanstrobl) [inline enrollment-server/src/main/resources/application.properties:74]: Missing space after `#`.

File: enrollment-server/src/main/resources/application.properties [hunk 5]
```diff
@@ -68,6 +68,16 @@ powerauth.service.security.clientSecret=
...
+spring.security.user.name=
+spring.security.user.password=
+
+#OIDC Settings
```
Comment (romanstrobl) [inline enrollment-server/src/main/resources/application.properties:78]: Missing space after `#`.

**Review by romanstrobl — APPROVED** [review]
Looks OK, I have few minor comments.

**Review by petrdvorak — CHANGES_REQUESTED** [review]
I am strongly against any admin endpoints for Enrollment Server.

**Review by petrdvorak — APPROVED** [review]
I really do not like the solution. Let's merge it for this release and look for improvements in next versions.

**Pull request merged**